R/tax_check.R

```diff
@@ -0,0 +1,320 @@
+  }
+  x[grep("^$|^\\s+$", x[,names]),names] <- NA
+  if(!is.character(x[,names]) | all(is.na(x[,names]))) {
+    stop("The 'names' column in 'x' must contain data of class character and
+         at least one entry that is not NA or empty")
+  }
+
+  # groups: If not NULL, a 1L character vector denoting a character column in x
+  if(!is.null(groups)) {
+    if(any(c(!is.atomic(groups), length(groups) != 1,
+             !groups %in% colnames(x)))) {
+      stop("Please specify 'groups' as a single column name in 'x")
+    }
+    if(!is.character(x[,groups])) {
+      stop("The 'groups' column in 'x' must contain data of class character")
+    }
+    groups <- gsub("^$|^\\s+$", NA, x[,groups])
+  } else {
+    groups <- substring(x[,names], 1, 1)
+  }
+
+  # sim: a 1L numeric > 0 and < 1
+  if(any(c(!is.numeric(sim), length(sim) != 1, !is.atomic(sim)))) {
+    stop("'sim' must be a single numeric, greater than 0, less than 100")
+  }
+  if(sim >= 100 | sim <= 0) {
+    stop("'sim' must be a single numeric, greater than 0, less than 100")
+  }
+
+  # start: a 1L integer >= 0
+  if(!is.null(start)) {
+    if(any(c(!is.numeric(start), length(start) != 1, !is.atomic(start)))) {
+      stop("'start' must be a single positive integer, or zero")
+    }
+    if(any(c(start < 0, start %% 1 != 0, is.nan(start), is.infinite(start)))) {
+      stop("'start' must be a single positive integer, or zero")
+    }
+  }
+
+  # end: a 1L integer >= 0
+  if(!is.null(end)) {
+    if(any(c(!is.numeric(end), length(end) != 1, !is.atomic(end)))) {
+      stop("'end' must be a single positive integer, or zero")
+    }
+    if(any(c(end < 0, end %% 1 != 0, is.nan(end), is.infinite(end)))) {
+      stop("'end' must be a single positive integer, or zero")
+    }
+  }
+
+  # pref, suff: If not NULL, character vectors
+  if(is.null(pref)) {pref <- ""}
+  if(is.null(suff)) {suff <- ""}
+  if(any(c(!is.atomic(c(pref, suff)), !is.character(pref),
+           !is.character(suff)))) {
+    stop("'pref' and 'suff' must be character vectors or NULL")
+  }
+
+  # verbose: a 1L logical vector
+  if(any(c(!is.atomic(c(verbose)), !is.logical(verbose),
+           length(verbose) != 1))) {
+    stop("'verbose' must be a single logical value")
+  }
+
+  # check for non-letter characters, returning NULL if none
+  nm <- as.integer(grep("[^[:alpha:] ]", x[,names]))
+  if(length(nm) != 0) {
+    if(verbose) {warning("Non-letter characters present in the taxon names")}
+  } else {
+    nm <- NULL
+  }
+
+
+  # FORMAT INPUT DATA ------------------------------------------------------- #
+
+  # names data.frame, drop missing names, fill missing groups alphabetically
+  x <- x2 <- data.frame(groups = groups, names = x[,names])
+  x <- x[!duplicated(x),,drop = FALSE]
+  x <- x[!is.na(x[,"names"]),,drop = FALSE]
+  no_group <- which(is.na(x[,"groups"]))
+  x[no_group, "groups"] <- substring(x[no_group, "names"], 1, 1)
+  # convert from similarity to dissimilarity
+  sim <- 1 - (sim / 100)
```

Comment:
I didn't really intend for you to change this from dissimilarity to similarity, just more to clarify that if percentages are desired it should be given as a percentage (whether similarity or dissimilarity). It is fine as similarity but we should make this conversion clear in the details as the the Jaro metric returns dissimilarity?